python/mxnet/io.py

```diff
@@ -615,13 +629,13 @@ def __init__(self, data, label=None, batch_size=1, shuffle=False,
                  label_name='softmax_label'):
         super(NDArrayIter, self).__init__(batch_size)
 
-        self.data = _init_data(data, allow_empty=False, default_name=data_name)
-        self.label = _init_data(label, allow_empty=True, default_name=label_name)
-        if isinstance(data, CSRNDArray) or isinstance(label, CSRNDArray):
+        if _has_instance(data, CSRNDArray) or _has_instance(label, CSRNDArray):
```

Comment:
Thanks for the fix. Shall we have a unit test with invalid inputs and assert Exception?

Reply:
Added :)